src/components/calcite-picker-item/calcite-picker-item.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  checkSquare16,
+  circle16F,
+  circleFilled16F,
+  handleVertical24,
+  square16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import { ICON_TYPES } from "../calcite-picker/resources";
+import CalciteIcon from "../utils/CalciteIcon";
+import { getElementDir } from "../utils/dom";
+
+@Component({
+  tag: "calcite-picker-item",
+  styleUrl: "./calcite-picker-item.scss",
+  shadow: true
+})
+export class CalcitePickerItem {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) editing = false;
+
+  @Prop() selected = false;
+
+  @Watch("selected")
+  selectedWatchHandler(newValue) {
+    if (this.isSelected !== newValue) {
+      this.isSelected = newValue;
+      this.emitChangeEvent();
+    }
+  }
+
+  @Prop({ reflect: true }) icon: ICON_TYPES | null = null;
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() isSelected = this.selected;
+
+  dir: "rtl" | "ltr";
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.dir = getElementDir(this.el);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerItemSelectedChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggleSelected(coerce?: boolean, emit = false) {
+    this.isSelected = typeof coerce === "boolean" ? coerce : !this.isSelected;
+    if (emit) {
+      this.emitChangeEvent();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  pickerClickHandler = (event: MouseEvent): void => {
+    this.isSelected = !this.isSelected;
+    this.emitChangeEvent(event.shiftKey);
+  };
+
+  secondaryActionContainerClickHandler(event: MouseEvent) {
+    event.stopPropagation();
+  }
+
+  emitChangeEvent(shiftPressed = false) {
+    this.calcitePickerItemSelectedChange.emit({
+      item: this.el,
+      value: this.value,
+      selected: this.isSelected,
+      shiftPressed
```

Comment:
It doesn't hurt anything to have the shiftKey here, but if the click handler was on the `PIcker` it would be unnecessary to pass it via event.

Reply:
The Picker is what needs to know if a shift key was pressed, so it should be listening to the click event instead of relying on that event information to be passed via a child component.

Reply:
This is a stylistic thing that has no impact beyond it being different. Closing out this thread, please open a new issue if you want to discuss further.

Reply:
Its an architectural difference. But if we decided that control clicking should also do the same as shift, it would require a change on the custom event. It would be cleaner if the event information didn't have to be passed to the picker.